src/python_testing/TC_TLSCERT.py

```diff
@@ -0,0 +1,569 @@
+        asserts.assert_equal(found_certs[response.caid], first_cert, "Expected matching certificate detail")
+
+        self.step(10)
+        if not self.skip_cr2:
+            attribute_certs = await cr2_cmd.read_tls_cert_attribute(attribute=attributes.ProvisionedRootCertificates)
+            asserts.assert_greater_equal(len(attribute_certs), 1, "Expected at least 1 certificate")
+            found_certs = dict()
+            # Python controller is using TCP, which supports large payloads; need a way to test
+            # over MRP that certificate is not present
+            for cert in attribute_certs:
+                found_certs[cert.caid] = cert.certificate
+            asserts.assert_in(response2.caid, found_certs, "ProvisionedRootCertificates should contain provisioned root cert")
+            asserts.assert_equal(found_certs[response2.caid], second_cert, "Expected matching certificate detail")
+
+        self.step(11)
+        find_response = await cr1_cmd.send_find_root_command(caid=response.caid)
+        asserts.assert_equal(len(find_response.certificateDetails), 1, "Expected single certificate")
+        asserts.assert_equal(find_response.certificateDetails[0].certificate, first_cert)
+
+        self.step(12)
+        if not self.skip_cr2:
+            find_response2 = await cr2_cmd.send_find_root_command(caid=response2.caid)
+            asserts.assert_equal(len(find_response2.certificateDetails), 1, "Expected single certificate")
+            asserts.assert_equal(find_response2.certificateDetails[0].certificate,
+                                 second_cert, "Expected matching certificate detail")
+
+        self.step(13)
+        find_all_response = await cr1_cmd.send_find_root_command()
+        asserts.assert_greater_equal(len(find_all_response.certificateDetails), 1, "Expected at least 1 certificate")
+        found_certs = dict()
+        for cert in find_all_response.certificateDetails:
+            found_certs[cert.caid] = cert.certificate
+        asserts.assert_in(response.caid, found_certs, "FindRootCertificate should contain provisioned root cert")
+        asserts.assert_equal(found_certs[response.caid], first_cert, "Expected matching certificate detail")
+
+        self.step(14)
+        if not self.skip_cr2:
+            find_all_response = await cr2_cmd.send_find_root_command()
+            asserts.assert_greater_equal(len(find_all_response.certificateDetails), 1, "Expected at least 1 certificate")
+            found_certs = dict()
+            for cert in find_all_response.certificateDetails:
+                found_certs[cert.caid] = cert.certificate
+            asserts.assert_in(response2.caid, found_certs, "FindRootCertificate should contain provisioned root cert")
+            asserts.assert_equal(found_certs[response2.caid], second_cert, "Expected matching certificate detail")
+
+        self.step(15)
```

Comment:
Based on the test plan, some of steps where the StatusCode of NotFound is returned for FindRootCertificate and ProvisionRootCertificate need to be checked. 
These can be tackled as an update to the test script in a subsequent PR.

Reply:
Yup exactly, my plan was to make sure we have a good base here (end-to-end coverage of CRUD), then get the test plan finalized, then once it's finalized & all outstanding comments in test plan review are addressed, finalize the script (so we don't have to keep iterating on the script to match plan)